src/store.js

```diff
@@ -45,13 +45,17 @@ class Store {
    * does the reaction and returns the new state
    */
   handle(state, type, payload) {
+    var result = state
     var handler = this.__handlers.get(type)
 
     if (typeof handler === 'function') {
-      return handler.call(this, state, payload, type)
+      result = handler.call(this, state, payload, type)
+
+      if (result === undefined)
+        throw new Error("Store handler must return a value, did you forget a return statement")
```

Comment:
minor: I think the coding style for this project is to use braces around if blocks (aside from 1-liners)